internal/v1/api.yaml

```diff
@@ -1302,3 +1112,460 @@ components:
+    Customizations:
+      type: object
+      properties:
+        containers:
+          type: array
+          items:
+            $ref: '#/components/schemas/Container'
+            description: Container images to embed into the final artfact
+        directories:
+          type: array
+          items:
+            $ref: '#/components/schemas/Directory'
+            description: Directories to create in the final artifact
+        files:
+          type: array
+          items:
+            $ref: '#/components/schemas/File'
+            description: Files to create in the final artifact
+        subscription:
+          $ref: '#/components/schemas/Subscription'
+        packages:
+          type: array
+          maxItems: 10000
+          example: ['postgresql']
+          items:
+            type: string
+        payload_repositories:
+          type: array
+          items:
+            $ref: '#/components/schemas/Repository'
+        custom_repositories:
+          type: array
+          items:
+            $ref: '#/components/schemas/CustomRepository'
+        openscap:
+          $ref: '#/components/schemas/OpenSCAP'
+        filesystem:
+          type: array
+          maxItems: 128
+          items:
+            $ref: '#/components/schemas/Filesystem'
+        users:
+          type: array
+          items:
+            $ref: '#/components/schemas/User'
+          description:
+            "list of users that a customer can add, also specifying their respective groups and SSH keys"
+        services:
+            $ref: '#/components/schemas/Services'
+        hostname:
+          type: string
+          description: Configures the hostname
+          example: myhostname
+        kernel:
+          $ref: '#/components/schemas/Kernel'
+        groups:
+          type: array
+          description: List of groups to create
+          items:
+            $ref: '#/components/schemas/Group'
+        timezone:
+          $ref: '#/components/schemas/Timezone'
+        locale:
+          $ref: '#/components/schemas/Locale'
+        firewall:
+          $ref: '#/components/schemas/FirewallCustomization'
+        installation_device:
+          type: string
+          description: |
+            Name of the installation device, currently only useful for the edge-simplified-installer type
+          example: /dev/sda
+        fdo:
+          $ref: '#/components/schemas/FDO'
+        ignition:
+          $ref: '#/components/schemas/Ignition'
+        partitioning_mode:
+          type: string
+          enum:
+            - raw
+            - lvm
+            - auto-lvm
+          description: |
+            Select how the disk image will be partitioned. 'auto-lvm' will use raw unless
+            there are one or more mountpoints in which case it will use LVM. 'lvm' always
+            uses LVM, even when there are no extra mountpoints. 'raw' uses raw partitions
+            even when there are one or more mountpoints.
```

Comment:
Do we want to add `fips` now too? Or do that separately?

Reply:
oh that's new!